IlmBase/IlmThread/IlmThreadWin32.cpp

```diff
@@ -38,11 +38,10 @@
 //
 //-----------------------------------------------------------------------------
 
-
 #include "IlmBaseConfig.h"
 
 #ifdef ILMBASE_FORCE_CXX03
-#ifdef _WIN32
+#if ((defined _WIN32 || defined _WIN64) && !defined(__MINGW64_VERSION_MAJOR))
```

Comment:
As a side note, we could also simply use the `__MINGW32__` marcro. See:
https://sourceforge.net/p/predef/wiki/Compilers/#mingw-and-mingw-w64

It does not need to have the `<windows.h>` included.

Reply:
__MINGW32__ is not appropriate because it's also present in the original MinGW and my PR is for Mingw-w64. 

I hadn't tested it with forcing of CXX03 mode so I guess that's something I have to fix. Why are you purposefully disabling c++11 though?

Reply:
> `__MINGW32__` is not appropriate because it's also present in the original MinGW and my PR is for Mingw-w64.

Ah, I see. Thanks for the clarification.

> Why are you purposefully disabling c++11 though?

I'm building without POSIX threads (MXE target: `x86_64-w64-mingw32.shared.win32`) because our library is somehow slower with POSIX threads functionality than with the native win32 implementation (see https://github.com/libvips/build-win64/pull/27 for the benchmarks). 

Building without POSIX threads has the disadvantage that the C++11 `<thread>`, `<mutex>`, and `<future>` implementations aren't available (see for example [this log](https://gist.github.com/kleisauke/c9c489a302057f35cd948266e412edd8)). So, unfortunately, I have to enable the CXX03 mode within OpenEXR.

Reply:
I see. I'll probably integrate your fixes whenever I get back to my computer.

Reply:
BTW I recently learned of https://github.com/mstorsjo/llvm-mingw, which use mingw windows headers and libs, but *not* the POSIX emulation. libc++ evidentially has direct windows support, and it might be faster too.

There's also clang + MSVC directly. I am working with the llvm-mingw author a bit to make a fully automated version cross toolchain for that in https://github.com/NixOS/nixpkgs/pull/72366. (Fully automated in that it downloads VS, and builds all the packages.)

Reply:
@Ericson2314 Thanks for the suggestion regarding https://github.com/mstorsjo/llvm-mingw! I've integrated it into our MXE-based build environment (https://github.com/libvips/build-win64-mxe/commit/f8c17ee55a75222bc34f1dd63bcea8986b652e93) and I'm considering making a PR at [MXE](https://github.com/mxe/mxe) to integrate it there too.